Add chrome extension docs (#1004)
clerk/clerk-docs: docs/references/chrome-extension/overview.mdx

---
title: Chrome Extension SDK
description: Learn how to integrate Clerk into your Chrome Extension
---

# Clerk Chrome Extension SDK

## Chrome Extension Key

> A Chrome Extension can be identified by its unique, in a similar way to how a website can be identified by its domain. You will need to explicitly configure your Chrome Extension's key or it will change often. If the key changes it can cause the Chrome Extension to fail.

Reviewer: ```suggestion
A Chrome Extension can be identified by its unique ID, in a similar way to how a website can be identified by its domain. You will need to explicitly configure your Chrome Extension's key or it will change often. If the key changes it can cause the Chrome Extension to fail.
```

---
title: Chrome Extension SDK
description: Learn how to integrate Clerk into your Chrome Extension
---

Clerk provides a Chrome Extension SDK that allows you to easily integrate Clerk into your Chrome Extension. This SDK provides a simple way to authenticate users and manage their sessions.

## Authenication Options

The authentication options for a Chrome Extension will differ depending on how the extension is configured. Please consult the following table and information to determine the option available.

- **Popup** - The default popup behaviour of a Chrome Extension
- **Side Panel** - A Chrome Extension configured and opened in a Side Panel

> - **Sync Host** - A Chrome Extension deployed along with a web application, and use using Sync Host to sync the auth state from the web application to the extension.

Reviewer 2: ```suggestion
- **Sync Host** - A Chrome Extension deployed along with a web application using Sync Host to sync the auth state from the web application to the extension.
```

Author: Fixed

---
title: Chrome Extension SDK
description: Learn how to integrate Clerk into your Chrome Extension
---

Clerk provides a Chrome Extension SDK that allows you to easily integrate Clerk into your Chrome Extension. This SDK provides a simple way to authenticate users and manage their sessions.

## Authentication options

When creating your Clerk application in the Clerk Dashboard, your authentication options will depend on how you configure your Chrome Extension. You can configure your Chrome Extension to behave as one of the following:

- **Popup** - Opens as a popup. This is the default behavior.
- **Side Panel** - Opens in a side panel
- **Sync Host** - Deployed alongside a web app that uses [Sync Host](/docs/references/chrome-extension/sync-host).

See the following table to determine the authentication options available for each configuration.

| | Popup | Side Panel | Sync Host |
| - | - | - | - |
| Email + OTP | ✅ | ✅ | ✅ |
| Email + Link | | | ✅ |
| Email + Password | ✅ | ✅ | ✅ |
| Username + Password | ✅ | ✅ | ✅ |
| SMS + OTP | ✅ | ✅ | ✅ |
| OAuth | | | ✅ |
| Google One Tap | | | ✅ |
| SAML | | | ✅ |
| Passkeys | ✅ | ✅ | ✅ |

> | Web3 | | | |

Reviewer: ```suggestion
| Web3 | | | ✅ |
```

> [!WARNING]
> Our Chrome Extension SDK currently does not fully support Sync Host on side panels. Currently, if a user authenticates in your web app, they need to close and reopen the side panel to update their auth status.

## Configure a consistent CRX ID

A Chrome Extension can be identified by its unique CRX ID, similar to how a website can be identified by its domain. The CRX ID rotates by default, which can cause errors with the Clerk integration. [Learn how to configure a consistent CRX ID](/docs/references/chrome-extension/configure-consistent-crx-id) so that your extension will have a stable, unchanging key.

## Sync your Chrome Extension with your web app

Clerk allows you to sync the authentication state from your web app to your Chrome Extension using the Sync Host feature. When a user authenticates in your web app, they will also be authenticated in your Chrome Extension. See [the dedicated guide](/docs/references/chrome-extension/sync-host) for more information.

## `createClerkClient()`

> It's recommended to use `createClerkClient()` for Chrome Extension's that are configured as popups.

Reviewer 3: ```suggestion
It's recommended to use `createClerkClient()` for Chrome Extension's that need to interact with Clerk in a content script.
```

| Web3 | | | ✅ |

> [!WARNING]
> Our Chrome Extension SDK currently does not fully support Sync Host on side panels. Currently, if a user authenticates in your web app, they need to close and reopen the side panel to update their auth status.

> ## Configure a consistent CRX ID

Author: I think we should change the order of these? This was initially established when the CRX ID guide was required. Maybe the following order?

1. React Router 
2. Sync Host
3. `createClerkClient()`
4. Deploy to production
5. Configure a CRX ID

Reviewer 3: can do

Reviewer 3: updated! [b089102](https://github.com/clerk/clerk-docs/pull/1004/commits/b089102b7f14df60439d79fd7c6313e1a735050a)

A Chrome Extension can be identified by its unique CRX ID, similar to how a website can be identified by its domain. The CRX ID rotates by default, which can cause errors with the Clerk integration. [Learn how to configure a consistent CRX ID](/docs/references/chrome-extension/configure-consistent-crx-id) so that your extension will have a stable, unchanging key.

> ## Sync your Chrome Extension with your web app

Author: `## Sync auth status between your Chrome Extension and web app`

The title should probably mention something about auth or auth state.

Reviewer 3: can do!

Reviewer 3: updated! [b089102](https://github.com/clerk/clerk-docs/pull/1004/commits/b089102b7f14df60439d79fd7c6313e1a735050a)

Clerk allows you to sync the authentication state from your web app to your Chrome Extension using the Sync Host feature. When a user authenticates in your web app, they will also be authenticated in your Chrome Extension. See [the dedicated guide](/docs/references/chrome-extension/sync-host) for more information.

## `createClerkClient()`

> It's recommended to use `createClerkClient()` for Chrome Extension's that are configured as popups.

Reviewer 3: ```suggestion
It's recommended to use `createClerkClient()` for Chrome Extension's that need to interact with Clerk in a content script.
```

> [!WARNING]
> Our Chrome Extension SDK currently does not fully support Sync Host on side panels. Currently, if a user authenticates in your web app, they need to close and reopen the side panel to update their auth status.

## Sync auth status between your Chrome Extension and web app

Clerk allows you to sync the authentication state from your web app to your Chrome Extension using the Sync Host feature. When a user authenticates in your web app, they will also be authenticated in your Chrome Extension. See [the dedicated guide](/docs/references/chrome-extension/sync-host) for more information.

## `createClerkClient()`

It's recommended to use `createClerkClient()` for Chrome Extension's that need to interact with Clerk in a content script.

> The `createClerkClient()` helper initializes a new Clerk instance on demand and refreshes the session token if there is a valid, signed-in user. It can be used in a [content script](https://developer.chrome.com/docs/extensions/develop/concepts/content-scripts) or a [background service worker](https://developer.chrome.com/docs/extensions/develop/concepts/service-workers/basics) to access a user's information or session token. [Learn more about `createClerkClient()`](/docs/references/chrome-extension/create-clerk-client).

Author: `createClerkClient()` can only be used in a service worker.
